### ISSUE TITLE
False positive for directives_ordering

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#440]
I have a lint for the following code:

```
export 'basic_types.dart' show Point, Offset, Size;
export 'edge_insets.dart' show EdgeInsets;
```

/cc @JPaulsen

**Comment by JPaulsen** [comment]
@a14n can you show me all your directives in that file?

**Comment by a14n** [comment]
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/painting/decoration.dart#L13-L14

**Comment by JPaulsen** [comment]
I got the problem, if you follow [these rules](https://www.dartlang.org/guides/language/effective-dart/style#ordering), a package export and a relative export should be in the same block, so the directives_ordering rules thinks this block should be together:

```
export 'package:flutter/services.dart' show ImageConfiguration;

export 'basic_types.dart' show Point, Offset, Size;
export 'edge_insets.dart' show EdgeInsets;
```

then the 2nd import is lower alphabetically than the first one, so it is a lint.

We will discuss how to proceed, and fix the bug, thank you @a14n :)

**Comment by bwilkerson** [comment]
Unless there's a guideline to the contrary, export directives should be grouped exactly like import directives and then sorted within groups.

**Comment by JPaulsen** [comment]
Perfect, we will rewrite the rule to do that.

**Comment by a14n** [comment]
I still see the same issue: a lint on the same file at the same linehttps://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/painting/decoration.dart#L13
